pkgs/applications/science/electronics/openboardviewer/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchurl, fetchFromGitHub, pkgs }:
+
+stdenv.mkDerivation rec {
+  pname = "OpenBoardView";
+  version = "8.0";
+  glad_path = "../src/glad";
+
+  src = fetchFromGitHub {
+    owner = "OpenBoardView";
+    repo = pname;
+    rev = "45edcee7e4c67cd90522d070431c920214002027";
+    sha256 = "1wxr8c8zyxxp4addww6ml5a0pyknvb4lph1hd0lhzx4asjwjnz46";
+    fetchSubmodules = true;
+  };
+
+  src_egl = fetchurl {
+    url = "https://raw.githubusercontent.com/KhronosGroup/EGL-Registry/649981109e263b737e7735933c90626c29a306f2/api/egl.xml";
+    sha256 = "be5beb0bc9278d720730ca5f22df2d4a979a322075ee77593b49ace257953137";
+  };
+
+  src_khrplatform = fetchurl {
+    url = "https://raw.githubusercontent.com/KhronosGroup/EGL-Registry/649981109e263b737e7735933c90626c29a306f2/api/KHR/khrplatform.h";
+    sha256 = "e206a6931f98ffe1c5c7ece69c4f94bbe1c9279243f40cbe7782848a0d3fa2de";
+  };
+
+  src_gl = fetchurl {
+    url = "https://raw.githubusercontent.com/KhronosGroup/OpenGL-Registry/e82287b71f66a88194f4a39e35fdac83dbf72707/xml/gl.xml";
+    sha256 = "32517f41b05e0de322439f78ca8aec4bb9415ab3f93823f51b55cac93f684b4f";
+  };
+
+  src_glx = fetchurl {
+    url = "https://raw.githubusercontent.com/KhronosGroup/OpenGL-Registry/e82287b71f66a88194f4a39e35fdac83dbf72707/xml/glx.xml";
+    sha256 = "4f00f20e507c353e8cc7765cd414b659a53628a12c5c22459afdc7ddd0f620b6";
+  };
+
+  src_wgl = fetchurl {
+    url = "https://raw.githubusercontent.com/KhronosGroup/OpenGL-Registry/e82287b71f66a88194f4a39e35fdac83dbf72707/xml/wgl.xml";
+    sha256 = "4f36192884ce3e95b3a27d4ae1763fa57aba677fd31b421f8c496140f17538b6";
+  };
+
+  buildInputs = with pkgs; [
+    (SDL2.override { withStatic = true; }).dev
+    gnumake cmake python3Full pkg-config zlib
```

Comment:
`cmake`, `pkg-config`, `gcc10`, should go in `nativeBuildInputs`. Roughly speaking, they are build-time rather than runtime dependencies.

Reply:
If I put these into nativeBuildInputs it doesn't build anymore.

Reply:
Why not?

Reply:
also gnumake should be available by default and also belongs into nativeBuildInputs.

Reply:
At the moment, I am not able to build it anymore. I am getting the same errors as when I moved these packages to nativeBuildInputs. I will try to investigate that further.

```
/build/source/src/openboardview/FileFormats/BRDFile.cpp:216:37: error: expected unqualified-id before '&' token
  216 |    } catch (const std::out_of_range &e) {
      |                                     ^
/build/source/src/openboardview/FileFormats/BRDFile.cpp:216:36: error: expected ')' before '&' token
  216 |    } catch (const std::out_of_range &e) {
      |            ~                       ^~
      |                                    )
/build/source/src/openboardview/FileFormats/BRDFile.cpp:216:37: error: expected '{' before '&' token
  216 |    } catch (const std::out_of_range &e) {
      |                                     ^
/build/source/src/openboardview/FileFormats/BRDFile.cpp:216:38: error: 'e' was not declared in this scope
  216 |    } catch (const std::out_of_range &e) {
```

Reply:
After some investigation, the issue was a too new GCC version. GCC 10 causes that error, GCC 9 works fine.